File: Numbers/Reverse_Int/reverse_int.go
package main

import (
	"fmt"
	"math"
)

func main() {
	fmt.Println(reverseInt(123) == 321)
	fmt.Println(reverseInt(500) == 5)
	fmt.Println(reverseInt(503) == 305)
}

func reverseInt(n int32) int {
	n2 := int(math.Abs(float64(n)))
	result := 0
	for n2 > 0 {
		result = result*10 + n2%10
		n2 /= 10
	}
	if n < 0 {
		return -result
	}
	return result
}

File: Numbers/Reverse_Int/reverse_int.py

def reverse_int(n):
    n2 = abs(n)
    result = 0
    while n2 > 0:
        result = result * 10 + n2 % 10
        n2 /= 10
    return -result if n < 0 else result

print(reverse_int(123) == 321)
print(reverse_int(500) == 5)
print(reverse_int(503) == 305)
